src/main/java/ro/msg/learning/shop/configuration/LocationConfiguration.java

```diff
@@ -28,7 +38,20 @@ public LocationStrategy locationStrategy(@Value("${strategy.type}") String strat
             case "ABUNDANT" -> {
                 return new MostAbundantStrategy(stockRepository, stockMapper);
             }
+            case "GREEDY" ->
+            {
+                return new GreedyStrategy(locationRepository, stockRepository, stockMapper);
+            }
             default -> throw new StrategyNotImplementedException("Error. No location strategy implemented");
         }
     }
+
+    @Bean
+    public RestTemplate restTemplate(RestTemplateBuilder builder) {
+        return builder
+                .uriTemplateHandler(new DefaultUriBuilderFactory("http://www.mapquestapi.com/directions/v2/"))
```

Comment:
Please add this URL as a property in application.properties file. In a real project, you will never use hardcoded urls or any values, you will use either properties, environment variables or enums.